### PR TITLE
fix: don't error out if serve/chat model don't match

### PR DESCRIPTION
**Pull request by jaideepr97**
We supply a default model to `ilab model chat`. If this does not match the model being served separately (if there is one), instead of erroring out we should inform the user that we are ignoring what they supplied in favor of what is being served 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/chat.py [hunk 1]
```diff
@@ -781,14 +781,13 @@ def chat_cli(
...
-        if model == "merlinite-7b-lab-Q4_K_M":
-            logger.info(
-                f"Model {model} is not a full path. Try running ilab config init or edit your config to have the full model path for serving, chatting, and generation."
-            )
-        raise ChatException(
-            f"Model {model} is not served by the server. These are the served models: {model_ids}"
+    # if a model is already being served, ignore whatever model may have been supplied with the chat command
+    if len(model_ids) > 0 and not any(model == m for m in model_ids):
+        logger.info(
+            f"Requested model {model} is not served by the server. Proceeding to chat with served model: {model_ids}"
```
Comment (nathan-weinberg) [inline src/instructlab/model/chat.py:787]: Should this be `{model_ids[0]}`?

File: src/instructlab/model/chat.py [hunk 2]
```diff
@@ -781,14 +781,13 @@ def chat_cli(
...
         model_ids.append(m.id)
-    if not any(model == m for m in model_ids):
-        if model == "merlinite-7b-lab-Q4_K_M":
-            logger.info(
-                f"Model {model} is not a full path. Try running ilab config init or edit your config to have the full model path for serving, chatting, and generation."
-            )
-        raise ChatException(
-            f"Model {model} is not served by the server. These are the served models: {model_ids}"
+    # if a model is already being served, ignore whatever model may have been supplied with the chat command
+    if len(model_ids) > 0 and not any(model == m for m in model_ids):
```
[the remark this answers is not in the document]
Comment (jaideepr97) [reply src/instructlab/model/chat.py]: done

File: src/instructlab/model/chat.py (hunk 1 above)
Comment (jaideepr97) [reply src/instructlab/model/chat.py]: updated

**Review by nathan-weinberg — APPROVED** [review]
LGTM

**Review by cdoern — APPROVED** [review]

**Comment by cdoern** [comment]
going to force merge this if only failure is macOS functional tests due to #2753

**Comment by jaideepr97** [comment]
> going to force merge this if only failure is macOS functional tests due to #2753

testing again

**Comment by cdoern** [comment]
@Mergifyio rebase

**Comment by cdoern** [comment]
should fix CI

**Comment by cdoern** [comment]
@Mergifyio backport release-v0.21

**Pull request merged**